src/hooks/usePokemonDetails.ts

```diff
@@ -0,0 +1,16 @@
+import { useQuery } from "react-query";
+import { PokemonDetails } from "../interfaces/pokemonDetails";
+import { PokemonItem } from "../interfaces/pokemonItem";
+import { getPokemonDetails } from "../services/pokemonDetailsService";
+
+const usePokemonDetails = (pokemon: PokemonItem | null) => {
```

Comment:
Te lo volevo dire gia' in un'altra PR, e non devi neanche affrontarlo in questa, ma penso che sia strano che `pokemon` possa essere `null` avrei preferito fosse opzionale (quindi potenzialmente `undefined` e non null).